metals/src/main/scala/scala/meta/internal/metals/FallbackMetalsLspService.scala

```diff
@@ -56,26 +59,64 @@ class FallbackMetalsLspService(
   buildServerPromise.success(())
   indexingPromise.success(())
 
+  val pauseables: Pauseable = Pauseable.fromPausables(
+    parseTrees ::
+      compilations.pauseables
+  )
+
   private val files: AtomicReference[Set[AbsolutePath]] = new AtomicReference(
     Set.empty
   )
 
-  override def maybeImportScript(path: AbsolutePath): Option[Future[Unit]] = {
-    if (!path.isScala) None
-    else {
-      val prev = files.getAndUpdate(_ + path)
-      if (prev.contains(path)) None
-      else Some(scalaCli.start(path))
-    }
-  }
-
   override def didClose(params: DidCloseTextDocumentParams): Unit = {
     val path = params.getTextDocument.getUri.toAbsolutePath
     files.getAndUpdate(_ - path)
     super.didClose(params)
     scalaCli.stop(path)
   }
 
+  protected def buildData(): Seq[BuildTool] =
+    scalaCli.lastImportedBuilds.map {
+      case (lastImportedBuild, buildTargetsData) =>
+        Indexer
+          .BuildTool("scala-cli", buildTargetsData, lastImportedBuild)
+    }
+  def buildHasErrors(path: AbsolutePath): Boolean = false
+  protected def check(): Unit = {}
+  protected def didCompileTarget(
+      report: ch.epfl.scala.bsp4j.CompileReport
+  ): Unit = {}
+  def fileWatcher: FileWatcher = NoopFileWatcher
+  def getTargetsInfoForReports(): List[Map[String, String]] = Nil
```

Comment:
Since this comes from the doctor it might be worth adding doctor as a explicit dependency to the FolderReportsZippper.